src/library_dylink.js

```diff
@@ -234,29 +234,29 @@ var LibraryDylink = {
   },
 
   // We support some amount of allocation during startup in the case of
-  // dynamic linking, which needs to allocate memory for dynamic libraries that
-  // are loaded. That has to happen before the main program can start to run,
+  // dynamic linking, which is needed to allocate the memory for the library's
+  // data segment. That has to happen before the main program can start to run,
   // because the main program needs those linked in before it runs (so we can't
   // use normally malloc from the main program to do these allocations).
 
-  // Allocate memory no even if malloc isn't ready yet.
-  $getMemory__deps: ['$GOT'],
-  $getMemory: function(size) {
+  // Allocate memory for the module's data segment even if malloc isn't ready yet.
+  $getOrAllocateModuleMemory__deps: ['$GOT'],
+  $getOrAllocateModuleMemory: function(size) {
     // After the runtime is initialized, we must only use sbrk() normally.
 #if DYLINK_DEBUG
-    err("getMemory: " + size + " runtimeInitialized=" + runtimeInitialized);
+    err("getOrAllocateModuleMemory: " + size + " runtimeInitialized=" + runtimeInitialized);
 #endif
     if (runtimeInitialized)
 #if USE_PTHREADS
-      // in a pthread the module heap was allocated in the main thread before
-      // the runtime was initialized
+      // In a pthread the module memory is allocated in the main thread before
+      // the runtime is initialized
```

Comment:
How about moving this check outside the outer one?

Can you add to the comment here something about like "because we only support load-time dynamic linking with pthreads today" and perhaps a TODO about redesigned this to use shared layout once we have runtime dynamic linking?